vllm/entrypoints/openai/serving_chat.py

```diff
@@ -127,6 +127,23 @@ def image_token_str(self) -> Optional[str]:
 
         return self.tokenizer.decode(image_token_id)
 
+    # TODO: Let user specify how to insert image tokens into prompt
+    # (similar to chat template)
+    def _get_full_image_text_prompt(self, image_token_str: str,
+                                    text_prompt: str) -> str:
+        """Combine image and text prompts for vision language model depending on
+        the model architecture."""
+
+        model_type = self.model_config.hf_config.model_type
+        if model_type in ("llava", "llava_next", "phi3_v"):
+            full_prompt = f"{image_token_str}\n{text_prompt}"
+        else:
+            # The model does not require image token in the prompt
+            # e.g, "blip-2", "chatglm"
+            return text_prompt
```

Comment:
In those cases image_token_str is None, so there should be no need to handle this case in this method.

Reply:
That's right - I think I should actually raise an error here if `image_token_str is not None` but the implementation for the model is missing.

Reply:
Hardcoding the model names would make it more difficult to add new models. I think it is fine to default to using newline character to separate image and text.

Reply:
Hmm that's fair - let me make it simple and clear about it then.